src/ffi/mod.rs

```diff
@@ -831,6 +844,72 @@ extern "C" {
     fn fortify_source();
 }
 
+#[cfg(feature = "hide_ips")]
+struct LogCensor {
+    ip_mask_seed: [u8; 32],
+    ip_masks: parking_lot::Mutex<HashMap<Ipv4Addr, String>>,
+    ip_regex: Regex,
+}
+
+#[cfg(feature = "hide_ips")]
+impl LogCensor {
+    fn new() -> Option<Self> {
+        Regex::new(
+            r"(?x)\b
+                (?:
+                    (?:
+                        25[0-5]
+                        |  2[0-4][0-9]
+                        |  1[0-9]{2}
+                        |  [1-9]?[0-9]
+                    )
+                    \.
+                ){3}
+                (?:
+                    25[0-5]
+                    |  2[0-4][0-9]
+                    |  1[0-9]{2}
+                    |  [1-9]?[0-9]
+                )
+                \b",
+        )
+        .ok()
+        .map(|re| LogCensor {
+            ip_masks: parking_lot::Mutex::new(HashMap::new()),
+            ip_mask_seed: rand::thread_rng().gen::<[u8; 32]>(),
+            ip_regex: re,
+        })
+    }
+
+    fn censor_logs(&self, log: String) -> String {
```

Comment:
This method should do nothing in debug mode (or there should be no instance of `LogCensor`).

Reply:
`LogCensor` doesn't exist in debug mode now

Reply:
Now it exists, but there is no instance of it